### PR TITLE
rustc: Remove support for int/uint

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit removes all parsing, resolve, and compiler support for the old and
long-deprecated int/uint types.

### CODE REVIEW

**Comment by aturon** [comment]
@bors: r+ 43bfaa4

Thanks for doing this!

**Comment by tshepang** [comment]
How is it that these didn't emit warnings when Rust was built?

File: src/libcore/num/f32.rs [hunk 1]
```diff
@@ -193,12 +193,12 @@ impl Float for f32 {
     #[inline]
     #[unstable(feature = "core")]
     #[deprecated(since = "1.0.0")]
-    fn mantissa_digits(_: Option<f32>) -> uint { MANTISSA_DIGITS as uint }
+    fn mantissa_digits(_: Option<f32>) -> usize { MANTISSA_DIGITS as usize }
```
Comment (nagisa) [inline src/libcore/num/f32.rs]: These shouldn’t be usize/isize.

**Comment by richo** [comment]
\o/

**Pull request merged**